lib/crawler/document_mapper.rb

```diff
@@ -68,11 +68,11 @@ def core_fields(crawl_result)
     def html_fields(crawl_result) # rubocop:disable Metrics/AbcSize
       remove_empty_values(
         title: crawl_result.document_title(limit: config.max_title_size),
-        body: crawl_result.document_body(limit: config.max_body_size),
+        body: crawl_result.document_body(limit: config.max_body_size, exclude_tags: config.exclude_tags),
         meta_keywords: crawl_result.meta_keywords(limit: config.max_keywords_size),
         meta_description: crawl_result.meta_description(limit: config.max_description_size),
         links: crawl_result.links(limit: config.max_indexed_links_count),
-        headings: crawl_result.headings(limit: config.max_headings_count),
+        headings: crawl_result.headings(limit: config.max_headings_count, exclude_tags: config.exclude_tags),
         full_html: crawl_result.full_html(enabled: config.full_html_extraction_enabled)
```

Comment:
`html_fields` are part of the ES document for indexing. Before I go any deeper, do we want everything here to be the version stripped out of excluded tags? If that's the case are we okay with keeping the pattern of passing the config as param, or would we want to make the config live in the `CrawlResult` object itself?

Reply:
That's a very fair point, I've removed the change from `headings` in [9825a06](https://github.com/elastic/crawler/pull/346/commits/9825a061832685ebd3fc5779b13c198ead85865e)and kept only in `body`, to keep it consistent with how data attributes exclusion does it. I'll give the refactoring a go in a separate PR so I don't blow the scope too much here, and see how it looks. :) Maybe the intention was to keep `CrawlResult` fairly naive and only make higher-level classes know about the config. 🤷‍♀️